api/api.go

```diff
@@ -806,3 +814,35 @@ func (b *BlockChainAPI) prepareBlockResponse(
 
 	return blockResponse, nil
 }
+
+func (b *BlockChainAPI) getCadenceHeight(
+	blockNumberOrHash *rpc.BlockNumberOrHash,
+) (uint64, error) {
+	if number, ok := blockNumberOrHash.Number(); ok {
+		height := uint64(0)
+		if number >= 0 {
+			var err error
+			height, err = b.blocks.GetCadenceHeight(uint64(number.Int64()))
+			if err != nil {
+				b.logger.Error().Err(err).Msg("failed to get cadence height")
+				return 0, err
+			}
+		}
+		return height, nil
+	} else if hash, ok := blockNumberOrHash.Hash(); ok {
+		block, err := b.blocks.GetByID(hash)
```

Comment:
maybe ok for now, but maybe add a todo that we can introduce a new index with block hash to block height, since this currently is a more expensive operation

Reply:
You are right, I might get to this in the current PR, so that we don't create more backlog :sweat_smile:

Reply:
It turns out that the index was already there, I simply added a helper function to retrieve it.
Updated in https://github.com/onflow/flow-evm-gateway/pull/192/commits/c2356f46cbcec51b4beba9c1f26390e80a8ef058